File: 04-SORTING/PHP/sort-colors.php
<?php


# https://leetcode.com/problems/sort-colors/
class Solution
{
    /**
     * Bucket Sorting Algorithm
     */
    public function sortColors(array &$nums): void
    {
        $counts = [0, 0, 0];

        foreach ($nums as $n) {
            $counts[$n]++;
        }

        $i = 0;
        for ($n = 0; $n < count($counts); $n++) {
            for ($j = 0; $j < $counts[$n]; $j++) {
                $nums[$i] = $n;
                $i++;
            }
        }
    }
}

File: 04-SORTING/Python/sort-colors.py
from typing import *


# https://leetcode.com/problems/sort-colors/
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Bucket Sorting Algorithm
        """
        counts = [0, 0, 0]

        for n in nums:
            counts[n] += 1

        i = 0
        for n in range(len(counts)):
            for _ in range(counts[n]):
                nums[i] = n
                i += 1

